be/src/storage/lake/gc.cpp

```diff
@@ -13,120 +13,155 @@
-    auto len = std::strftime(buffer, sizeof(buffer), "%Y.%m.%d %H:%M:%S UTC", gmtime_r(&ts, &tm));
-    return {buffer, len};
+static Status write_orphan_list_file(const std::set<std::string>& orphans, WritableFile* file) {
+    for (const auto& s : orphans) {
+        RETURN_IF_ERROR(file->append(s));
+        RETURN_IF_ERROR(file->append("\n"));
+    }
+    DCHECK_EQ((kSegmentFileNameLength + 1) * orphans.size(), file->size());
+    return file->close();
 }
 
-Status metadata_gc(std::string_view root_location, TabletManager* tablet_mgr, int64_t min_active_txn_id) {
-    ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(root_location));
+static Status read_orphan_list_file(RandomAccessFile* file, std::vector<std::string>* orphans) {
+    auto stream = file->stream();
+    auto size_or = stream->get_size();
+    if (size_or.status().is_not_found()) { // File does not exist
+        return Status::OK();
+    } else if (!size_or.ok()) {
+        return size_or.status();
+    }
+    auto size = *size_or;
+    DCHECK_EQ(0, size % (kSegmentFileNameLength + 1));
+    raw::RawVector<char> buff;
+    buff.resize(size); // TODO: streaming read
+    RETURN_IF_ERROR(file->read_fully(buff.data(), size));
+    orphans->reserve(size / (kSegmentFileNameLength + 1));
+    for (auto offset = 0L; offset < size; offset += (kSegmentFileNameLength + 1)) {
+        orphans->emplace_back(&buff[offset], kSegmentFileNameLength);
+        DCHECK_EQ('\n', buff[offset + kSegmentFileNameLength + 1]);
+    }
+    return Status::OK();
+}
 
+static Status delete_tablet_metadata(std::string_view root_location, const std::set<int64_t>& owned_tablets) {
+    ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(root_location));
     const auto max_versions = config::lake_gc_metadata_max_versions;
     if (UNLIKELY(max_versions < 1)) {
         return Status::InternalError("invalid config 'lake_gc_metadata_max_versions': value must be no less than 1");
     }
 
-    const auto metadata_root_location = join_path(root_location, kMetadataDirectoryName);
-    const auto txn_log_root_location = join_path(root_location, kTxnLogDirectoryName);
-
     std::unordered_map<int64_t, std::vector<int64_t>> tablet_metadatas;
-    std::vector<std::string> txn_logs;
     std::unordered_map<int64_t, std::unordered_set<int64_t>> locked_tablet_metadatas;
 
     auto start_time = std::time(nullptr);
-    {
-        auto iter_st = fs->iterate_dir(metadata_root_location, [&](std::string_view name) {
-            if (is_tablet_metadata(name)) {
-                auto [tablet_id, version] = parse_tablet_metadata_filename(name);
+    auto metadata_root_location = join_path(root_location, kMetadataDirectoryName);
+    auto iter_st = fs->iterate_dir(metadata_root_location, [&](std::string_view name) {
+        if (is_tablet_metadata(name)) {
+            auto [tablet_id, version] = parse_tablet_metadata_filename(name);
+            if (owned_tablets.count(tablet_id) > 0) {
```

Comment:
else if (owned_tablets.count(tablet_id) == 0) ?

Reply:
`owned_tablets.count(tablet_id) == 0` means the tablet is owned by another process and nothing needs to do here.